plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1881,6 +1866,76 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
+            String operation,
+            String nodeVersion,
+            String nodeId,
+            long readVersion)
+    {
+        return new CommitInfoEntry(
+                commitVersion,
+                createdTime,
+                session.getUser(),
+                session.getUser(),
+                operation,
+                ImmutableMap.of("queryId", session.getQueryId()),
+                null,
+                null,
+                "trino-" + nodeVersion + "-" + nodeId,
+                readVersion,
+                ISOLATION_LEVEL,
+                Optional.of(true));
+    }
+
+    @Override
+    public void setTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Optional<Object>> properties)
+    {
+        DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
+        ProtocolEntry currentProtocolEntry = getProtocolEntry(session, handle.getSchemaTableName());
+
+        Optional<Integer> readerVersion = Optional.empty();
+        if (properties.containsKey(READER_VERSION_PROPERTY)) {
+            readerVersion = Optional.of((int) properties.get(READER_VERSION_PROPERTY)
+                    .orElseThrow(() -> new IllegalArgumentException("The " + READER_VERSION_PROPERTY + " property cannot be empty")));
+            if (readerVersion.get() < currentProtocolEntry.getMinReaderVersion()) {
+                throw new TrinoException(INVALID_TABLE_PROPERTY, format(
+                        "%s cannot be downgraded from %d to %d", READER_VERSION_PROPERTY, currentProtocolEntry.getMinReaderVersion(), readerVersion.get()));
+            }
+        }
+
+        Optional<Integer> writerVersion = Optional.empty();
+        if (properties.containsKey(WRITER_VERSION_PROPERTY)) {
+            writerVersion = Optional.of((int) properties.get(WRITER_VERSION_PROPERTY)
+                    .orElseThrow(() -> new IllegalArgumentException("The " + WRITER_VERSION_PROPERTY + " property cannot be empty")));
+            if (writerVersion.get() < currentProtocolEntry.getMinWriterVersion()) {
+                throw new TrinoException(INVALID_TABLE_PROPERTY, format(
+                        "%s cannot be downgraded from %d to %d", WRITER_VERSION_PROPERTY, currentProtocolEntry.getMinWriterVersion(), writerVersion.get()));
+            }
+        }
+
+        long commitVersion = handle.getReadVersion() + 1;
+
+        ProtocolEntry protocolEntry = new ProtocolEntry(
+                readerVersion.orElse(currentProtocolEntry.getMinReaderVersion()),
+                writerVersion.orElse(currentProtocolEntry.getMinWriterVersion()));
+
+        try {
+            TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriter(session, handle.getLocation());
+
+            long createdTime = Instant.now().toEpochMilli();
+            transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(session, commitVersion, createdTime, SET_TBLPROPERTIES_OPERATION, nodeVersion, nodeId, 0));
```

Comment:
The zero at the end here should be `handle.getReadVersion()`

Reply:
I was following `addColumn`, `setTableComment`, `setColumnComment`. In these cases, we set `readVersion` to 0. Should these methods also use `handle.getReadVersion()`?

https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L1231

Reply:
I think they should. Unless @ebyhr @findinpath have other thoughts?

Reply:
Updated with `handle.getReadVersion()` value